From 72016f5c777271dfa4dddae6afd7a2982833ab43 Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Thu, 21 Mar 2024 11:20:11 +0200
Subject: [PATCH] [FSW-4100] pldm_bloader: fix makefile paths

Paths in makefile are adjusted to better support build for FS1.

Change-Id: I2693e16040fa3bc8ff25f44c9dd7dc82cacc3d30
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/app/pldm_bloader/Makefile b/app/pldm_bloader/Makefile
index 9823a37..0ec381a 100644
--- a/app/pldm_bloader/Makefile
+++ b/app/pldm_bloader/Makefile
@@ -2,11 +2,11 @@
 OBJCOPY = $(CROSS_COMPILE)objcopy
 BASE_DEV_NAME := $(shell echo $(HL_DEV) | cut -d'_' -f1)
 HL_DEV_SUBTYPE := $(shell echo $(HL_DEV) | cut -d'_' -s -f2,3)
-HL_DEV_ARCH := arch/$(BASE_DEV_NAME)/$(HL_DEV_SUBTYPE)
-HL_DEV_ARCH_COMMON := arch/$(BASE_DEV_NAME)/common
 PREBOOT_ROOT_DIR := ../..
+HL_DEV_ARCH := $(PREBOOT_ROOT_DIR)/arch/$(BASE_DEV_NAME)/$(HL_DEV_SUBTYPE)
+HL_DEV_ARCH_COMMON := $(PREBOOT_ROOT_DIR)/arch/$(BASE_DEV_NAME)/common
 
-include $(PREBOOT_ROOT_DIR)/$(HL_DEV_ARCH_COMMON)/Makefile
+include $(HL_DEV_ARCH_COMMON)/Makefile
 
 CFLAGS += -I$(SPECS_DIR)
 CFLAGS += -I$(SPECS_EMBEDDED_DIR)
@@ -14,26 +14,30 @@
 CFLAGS += -I$(SPECS_DIR)/$(BASE_DEV_NAME)/asic_reg
 CFLAGS += -Wall -Werror
 CFLAGS += -g -c
-CFLAGS += -I $(PREBOOT_ROOT_DIR)/include/ -I $(PREBOOT_ROOT_DIR)/ -I $(PREBOOT_ROOT_DIR)/lib/ -I $(PREBOOT_ROOT_DIR)/lib/common/ -I $(PREBOOT_ROOT_DIR)/$(HL_DEV_ARCH)/
+CFLAGS += -I $(PREBOOT_ROOT_DIR)/include/ -I $(PREBOOT_ROOT_DIR)/ -I $(PREBOOT_ROOT_DIR)/lib/ -I $(PREBOOT_ROOT_DIR)/lib/common/ -I $(HL_DEV_ARCH)/
 
 CFLAGS += -Os -fdata-sections -ffunction-sections
 
-include $(PREBOOT_ROOT_DIR)/$(HL_DEV_ARCH_COMMON)/pcie_phy_fw.mk
+include $(HL_DEV_ARCH_COMMON)/pcie_phy_fw.mk
 CFLAGS += -DPHY_FW_SIZE=$(PHY_FW_SIZE)
 
 SFLAGS := $(CFLAGS) -D__ASSEMBLY__
 
-PLDM_BLOADER_LD_SCRIPT := $(shell ls $(PREBOOT_ROOT_DIR)/$(HL_DEV_ARCH)/pldm_bloader.lds.S)
+PLDM_BLOADER_LD_SCRIPT := $(shell ls $(HL_DEV_ARCH)/pldm_bloader.lds.S)
 LDFLAGS := -g -nostdlib -static
 LDFLAGS += -Wl,--gc-sections
-LDFLAGS += -Wl,--just-symbols $(shell ls $(PREBOOT_ROOT_DIR)/$(HL_DEV_ARCH)/exports.elf)
+ARBOOT_EXPORTS_FILE := $(HL_DEV_ARCH)/exports.elf
+ifeq ($(shell test -e $(ARBOOT_EXPORTS_FILE) && echo -n yes),yes)
+LDFLAGS += -Wl,--just-symbols $(ARBOOT_EXPORTS_FILE)
+endif
+
 LDLIBS := -lgcc
 
 PLDM_BLOADER_SRCS_C := $(shell ls *.c)
 PLDM_BLOADER_SRCS_C += $(shell ls $(PREBOOT_ROOT_DIR)/lib/common/*.c)
-PLDM_BLOADER_SRCS_C += $(shell ls $(PREBOOT_ROOT_DIR)/$(HL_DEV_ARCH_COMMON)/common.c)
-PLDM_BLOADER_SRCS_C += $(shell ls $(PREBOOT_ROOT_DIR)/$(HL_DEV_ARCH)/common.c)
-PLDM_BLOADER_SRCS_C += $(shell ls $(PREBOOT_ROOT_DIR)/$(HL_DEV_ARCH)/$(HL_TARGET).c)
+PLDM_BLOADER_SRCS_C += $(shell ls $(HL_DEV_ARCH_COMMON)/common.c)
+PLDM_BLOADER_SRCS_C += $(shell ls $(HL_DEV_ARCH)/common.c)
+PLDM_BLOADER_SRCS_C += $(shell ls $(HL_DEV_ARCH)/$(HL_TARGET).c)
 PLDM_BLOADER_APP_S := $(shell ls crt0_rom.S)
 
 PLDM_BLOADER_OBJECTS := $(PLDM_BLOADER_SRCS_C:.c=.o)
diff --git a/arch/fs1/arc/Makefile b/arch/fs1/arc/Makefile
index 73f948a..36ba315 100644
--- a/arch/fs1/arc/Makefile
+++ b/arch/fs1/arc/Makefile
@@ -5,7 +5,7 @@
 CFLAGS += -I$(SPECS_H15_DIR)/fs1/
 CFLAGS += -I$(SPECS_H15_DIR)/fs1/asic_reg
 CFLAGS += -I$(SPECS_EMBEDDED_DIR)/fs/common
-CFLAGS += -I$(SPECS_EMBEDDED_DIR)/fs/$(SPECS_EMB_DEV)
+CFLAGS += -I$(SPECS_EMBEDDED_DIR)/fs/$(DEV_PREFIX)
 
 PPBOOT_SRCS_C := $(filter-out lib/common/module_i2c_slave.c,$(PPBOOT_SRCS_C))
 PPBOOT_SRCS_C := $(filter-out lib/common/pif_ts.c,$(PPBOOT_SRCS_C))
